### ISSUE TITLE
remark config is ignored when formatting

### ISSUE THREAD

**Issue by qyurila** [remarkjs/vscode-remark#123]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

vscode-remark v2.1.0

### Steps to reproduce

1. Create following `package.json` file and run `pnpm install` (or `npm install`):

`package.json`

```jsonc
{
	"devDependencies": {
		"remark": "^14.0.2",
		"remark-cli": "^11.0.0",
		"remark-preset-lint-markdown-style-guide": "^5.1.2"
	},
	"remarkConfig": {
		"plugins": [
			"remark-preset-lint-markdown-style-guide"
		]
	}
}
```

2. Create any markdown file e.g.

`test.md`

```md
- item 1
-  item 2
```

3. Run VSCode command '**Format Document**', or '**Format Document...**' (if present) and select '**remark**' to format a markdown file.

### Expected behavior

`test.md` should be formatted using configured options, resulting:

```md
- item 1
- item 2
```

### Actual behavior

`test.md` is formatted using remark's default options, resulting:

```md
*   item 1
*   item 2
```

...and 'vscode-remark' shows following warnings (which are parts of `remark-preset-lint-markdown-style-guide`):

```text
Marker style should be `-` remark-lint(unordered-list-marker-style)
Incorrect list-item indent: remove 2 spaces remark-lint(list-item-indent)
```

This means `vscode-remark` recognises `remarkConfig` but ignores them when formatting.

The same thing happens with `.remarkrc` (etc.) file.

I'm not sure if this is a bug or a missing feature, but I think the formatting feature meant be configurable regarding to #13.

### Runtime

Node v16

### Package manager

pnpm

### OS

Windows

**Comment by qyurila** [comment]
I found that it's `remark-cli` what ignores config when making output. Perhaps I was overlooking something. Sorry for bothering.

**Comment by qyurila** [comment]
Extra infomations I found (might wrong) for any future wonderers:

`remark-lint` has no auto-fix feature (currently): see remarkjs/remark-lint#82.

`--out` option of `remark` makes output file(s) based on `remarkConfig.settings` (which determines `remark-stringify`'s behaviour) only, not on any `remark-lint` presets and rules.

So, if you want to auto-format your `.md` files with this extension (or `remark * --out`) based on, let's say, `remark-preset-lint-markdown-style-guide`, you have to configure `remarkConfig.settings` options too even if it seems like duplicating.

**Comment by wooorm** [comment]
Yep, the lint rules are different from the formatting package. I believe there are issues open on improving this!
